src/handlers/xp.js

```diff
@@ -0,0 +1,110 @@
+const sc = require('sourcecred').default
+const fetch = require('node-fetch')
+const { log } = require('../utils')
+const parseMyCred = require('../parser/myCred')
+const { environment } = require('../environment')
+
+module.exports = async function score(message) {
+  const NodeAddress = sc.core.address.makeAddressModule({
+    name: 'NodeAddress',
+    nonce: 'N',
+    otherNonces: new Map().set('E', 'EdgeAddress'),
+  })
+
+  const targetParameter = parseMyCred(message.content)
+  const targetUserDiscordID = targetParameter.slice(3, targetParameter.length - 1)
+
+  if (isNaN(targetUserDiscordID)) {
+    return message.reply(
+      'command parsing failed. Please use the !ac help command to see how to use the requested command properly.',
+    )
+  }
+
+  const credAccounts = await (
+    await fetch(`https://raw.githubusercontent.com/${environment('GITHUB_LEDGER_FILE_PATH')}`,
+    )
+  ).json()
+
+  try {
+    const accounts = credAccounts.accounts
+    for (let i = 0; i < accounts.length; i++) {
```

Comment:
Instead of looping over everything and calling continue, you can just do accounts.find to first get the correct account and then calculate the cred info after that.

Reply:
Parsing updated with a `.find` method